src/app/patient-dashboard/patient-info/edit-contacts.component.ts

```diff
@@ -0,0 +1,131 @@
+  public display: boolean = false;
+  public patientPhoneNumber: number;
+  public alternativePhoneNumber: number;
+  public patnerPhoneNumber: number;
+  public nextofkinPhoneNumber: number;
+  constructor(private patientService: PatientService,
+              private personResourceService: PersonResourceService,
+              private patientInfoService: PatientInfoService) {
+  }
+  ngOnInit() {
+    this.getPatient();
+  }
+  public showDialog() {
+    this.display = true;
+  }
+  public getPatient() {
+    this.patientService.currentlyLoadedPatient.subscribe(
+      (patient) => {
+        if (patient) {
+          this.patient = patient;
+          this.patnerPhoneNumber = this.patient.person.patnerPhoneNumber;
+          this.patientPhoneNumber = this.patient.person.patientPhoneNumber;
+          this.alternativePhoneNumber =
+            this.patient.person.alternativePhoneNumber;
+          this.nextofkinPhoneNumber = this.patient.person.nextofkinPhoneNumber;
+        }
+      }
+    );
+  }
+  public saveAttribute() {
+    let person = {
+      uuid: this.patient.person.uuid
+      };
+    let personAttributePayload = {
+      attributes: [{
+        value:  this.patientPhoneNumber,
+        attributeType: '72a759a8-1359-11df-a1f1-0026b9348838'
+      }, {
+        value:  this.alternativePhoneNumber,
+        attributeType: 'c725f524-c14a-4468-ac19-4a0e6661c930'
+      }, {
+        value:  this.nextofkinPhoneNumber,
+        attributeType: 'a657a4f1-9c0f-444b-a1fd-445bb91dd12d'
+      }, {
+        value:  this.patnerPhoneNumber,
+        attributeType: 'b0a08406-09c0-4f8b-8cb5-b22b6d4a8e46'
+      }]
+    };
+    let payLoad = [];
+    for (let a of personAttributePayload.attributes) {
+
+      if (a.attributeType) {
+        let attr = this.getPersonAttributeByAttributeTypeUuid(a.attributeType);
+        if (attr === undefined ) {
+          attr = _.filter(attr, function(attribute){
+            return attribute !== undefined && attribute !== null;
+          });
+
+        }
+        if (attr && a.value === null || a.value.toString() === '') {
+          payLoad.push({uuid: attr.uuid, voided: true});
+        } else {
+          payLoad.push(a);
+        }
+
+      }
+    }
+    personAttributePayload.attributes = payLoad;
+    this.filterUndefinedUuidFromPayLoad(personAttributePayload.attributes);
+    this.personResourceService.saveUpdatePerson(person.uuid, personAttributePayload).subscribe(
+      (success) => {
+        if (success) {
+         let contacts = {
+           nextofkinPhoneNumber : this.nextofkinPhoneNumber,
+           patnerPhoneNumber : this.patnerPhoneNumber,
+           patientPhoneNumber : this.patientPhoneNumber,
+           alternativePhoneNumber : this.alternativePhoneNumber,
+           patientUuid: this.patient.person.uuid
+          };
+          this.patientInfoService.publishUpdatedContactsData(contacts);
+        }
+
+      },
+      (error) => {
+        console.log('error', error);
```

Comment:
Report error to the user. We also need some busy indication as this will not be instance with remote site where internet fluctuates.